src/sas/qtgui/Plotting/Slicers/SectorSlicer.py

```diff
@@ -171,10 +172,10 @@ def _post_data(self, nbins=None):
         item = self._item
         if self._item.parent() is not None:
             item = self._item.parent()
-        GuiUtils.updateModelItemWithPlot(item, new_plot, new_plot.id)
+        # GuiUtils.updateModelItemWithPlot(item, new_plot, new_plot.id)
 
-        self.base.manager.communicator.plotUpdateSignal.emit([new_plot])
-        self.base.manager.communicator.forcePlotDisplaySignal.emit([item, new_plot])
+        # self.base.manager.communicator.plotUpdateSignal.emit([new_plot])
+        # self.base.manager.communicator.forcePlotDisplaySignal.emit([item, new_plot])
```

Comment:
This breaks the base Sector interactor used in 2D plotting. No sector plots are generated, or, if they are, they are no longer accessible to the user. Maybe add a new variable to the method call called something like `generate_plots` that defaults to True and your method(s) could override that? These signals need to be emitted by default.

Reply:
I've tried to get the slicer to work with the uncommented code but I cant seem to get to work. I'll try again tomorrow.

Reply:
I'm suggesting something like this and when you call `getSlice()`

```
def _post_data(self, nbins=None, show_plots=True):
    ...
    if show_plots:
        GuiUtils.updateModelItemWithPlot(item, new_plot, new_plot.id)
        self.base.manager.communicator.plotUpdateSignal.emit([new_plot])
        self.base.manager.communicator.forcePlotDisplaySignal.emit([item, new_plot])
```

Reply:
Rewritten taking into account next comment exposing _post_data with captureSlicer method.